### ISSUE TITLE
 --rkey works ?

### ISSUE THREAD

**Issue by Noname400** [kanhavishva/KeyHunt-Cuda#22]
Please explain in detail how --rkey works

**Comment by Mehdi3123** [comment]
yes work. example  -r 5000  =  5.000.000.000 keys randomly searching.[ R :  10 ]

**Comment by Noname400** [comment]
how can I see it?
i used -r 10 and couldn't find any keys for puzzles 1-32

**Comment by Noname400** [comment]
where can i add this code? I'll add and compile myself. Help.

**Comment by Mehdi3123** [comment]
be patient  :)

./KeyHunt -t 1 -g --gpui 0 --gpux 256,256 -m addresses --coin BTC -b --range 1:FFFFFFFF -r 1000 -i puzzle_1-32_sort.bin

**Comment by astral2009** [comment]
> да работает. пример -r 5000 = 5.000.000.000 ключей случайный поиск. [R: 10]

COIN TYPE    : ETHEREUM
SEARCH MODE  : Multi Address
DEVICE       : CPU
CPU THREAD   : 10
SSE          : NO
RKEY         : 5000 Mkeys
MAX FOUND    : 65536
ETH ADDRESSES: eth1.bin
OUTPUT FILE  : Found.txt

[00:03:36] [CPU+GPU: 7.55 Mk/s] [GPU: 0.00 Mk/s] [C: 0.000000 %] [R: 0] [T: 1,629,181,952 (31 bit)] [F: 0]

**Comment by Mehdi3123** [comment]
> > да работает. пример -r 5000 = 5.000.000.000 ключей случайный поиск. [R: 10]
> 
> COIN TYPE : ETHEREUM
> SEARCH MODE : Multi Address
> DEVICE : CPU
> CPU THREAD : 10
> SSE : NO
> RKEY : 5000 Mkeys
> MAX FOUND : 65536
> ETH ADDRESSES: eth1.bin
> OUTPUT FILE : Found.txt
> 
> [00:03:36] [CPU+GPU: 7.55 Mk/s] [GPU: 0.00 Mk/s] [C: 0.000000 %] [R: 0] [T: 1,629,181,952 (31 bit)] [F: 0]

NO > for CPU   -r 20 / 100
          for GPU   -r 2000 / 10000

KeyHunt-Cuda v1.07

COIN TYPE    : ETHEREUM
SEARCH MODE  : Multi Address
DEVICE       : CPU
CPU THREAD   : 2
SSE          : NO
RKEY         : 50 Mkeys
MAX FOUND    : 65536
ETH ADDRESSES: 1.5M_eth_sort.bin
OUTPUT FILE  : Found.txt

Loading      : 100 %Loaded       : 1,500,002 Ethereum addresses

Bloom at 0x55ad5e872000
  Version    : 2.1
  Entries    : 3000004
  Error      : 0.0000010000
  Bits       : 86265640
  Bits/Elem  : 28.755175
  Bytes      : 10783205 (10 MB)
  Hash funcs : 20

[00:01:46] [CPU+GPU: 0.99 Mk/s] [GPU: 0.00 Mk/s] [C: 0.000000 %] [R: 2] [T: 105,988,096 (27 bit)] [F: 0]

**Comment by Noname400** [comment]
> потерпи :)
> 
> ./KeyHunt -t 1 -g --gpui 0 --gpux 256,256 -m адресов --coin BTC -b --range 1: FFFFFFFF -r 1000 -i puzzle_1-32_sort.bin

tell me how long to wait. this is very lacking in the program

**Comment by kanhavishva** [comment]
with the ```--rkey``` program generate random starting keys in given range for that thread after completing given MegaKeys interval.
for example: ```--rkey 20000```, here interval is 20000 MegaKeys, program will generate new random starting keys after scanning 20000 MegaKeys. it will do this again and again after 20000 MegaKeys completion from last cycle.

**Comment by MegaH01** [comment]
there is no option to save/resume progress incase of a PC crash ,etc.  If I am scanning for 6+ months would be nice to have a regular checkpoint file written that can be used to resume from where it left off
